cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx

```diff
@@ -103,6 +113,21 @@ export function AzureClusterSettingsStep(props: Partial<StepProps>) {
         setSelectedInstanceTypeId(instanceType);
     };
 
+    const onOsImageSelected = (imageName: string) => {
+        images.some((image) => {
+            if (image.name === imageName) {
+                setImageParameters(image);
+            }
+        });
+    };
+
+    useEffect(() => {
+        AzureService.getAzureOsImages().then((data) => {
```

Comment:
Should handle error state by providing a notification "Unable to retrieve a list of OS images: " and include the body of the error text.

Reply:
Done